backend/daemon/api/documents/v1alpha/comments.go

```diff
@@ -0,0 +1,343 @@
+			batch.Add(vcs.RootNode, mttdoc.AttrConvSelector, snode)
+		}
+
+		batch.Add(vcs.RootNode, mttdoc.AttrBlockSnapshot, encodeBlock(in.InitialComment))
+
+		dirty := batch.Dirty()
+
+		conn.AddDatoms(convo, change, dirty...)
+		conn.SaveVersion(convo, "main", meLocal, vcsdb.LocalVersion{change})
+		conn.EncodeChange(change, me.DeviceKey())
+
+		for _, d := range dirty {
+			if err := backlinks.IndexDatom(conn, convo, change, d); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return api.loadConversation(conn, perma.ID)
+}
+
+// AddComment implements the Comments server.
+func (api *Server) AddComment(ctx context.Context, in *documents.AddCommentRequest) (*documents.Block, error) {
+	me, err := api.me.Await(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if in.ConversationId == "" {
+		return nil, errutil.MissingArgument("conversationID")
+	}
+
+	convid, err := cid.Decode(in.ConversationId)
+	if err != nil {
+		return nil, errutil.ParseError("conversationID", in.ConversationId, convid, err)
+	}
+
+	if err := validateComment("comment", in.Comment); err != nil {
+		return nil, err
+	}
+
+	conn, release, err := api.vcsdb.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+
+	if err := conn.WithTx(true, func() error {
+		meLocal := conn.LookupIdentity(me)
+		obj := conn.LookupPermanode(convid)
+
+		clock := hlc.NewClock()
+		main := conn.GetVersion(obj, "main", meLocal)
+		for _, v := range main {
+			clock.Track(hlc.Unpack(conn.GetChangeMaxTime(obj, v)))
+		}
+
+		change := conn.NewChange(obj, meLocal, main, clock)
+		batch := vcs.NewBatch(clock, me.DeviceKey().Abbrev())
+
+		batch.Add(vcs.RootNode, mttdoc.AttrBlockSnapshot, encodeBlock(in.Comment))
+
+		dirty := batch.Dirty()
+		conn.AddDatoms(obj, change, dirty...)
+		conn.SaveVersion(obj, "main", meLocal, vcsdb.LocalVersion{change})
+		blk := conn.EncodeChange(change, me.DeviceKey())
+
+		if err := conn.Err(); err != nil {
+			return nil
+		}
+
+		in.Comment.Revision = blk.Cid().String()
+
+		for _, d := range dirty {
+			if err := backlinks.IndexDatom(conn, obj, change, d); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return in.Comment, nil
+}
+
+// DeleteConversation implements the Comments server.
+func (api *Server) DeleteConversation(context.Context, *documents.DeleteConversationRequest) (*emptypb.Empty, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method DeleteConversation not implemented")
+}
+
+// ResolveConversation implements the Comments server.
+func (api *Server) ResolveConversation(context.Context, *documents.ResolveConversationRequest) (*documents.ResolveConversationResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method ResolveConversation not implemented")
+}
+
+// DeleteComment implements the Comments server.
+func (api *Server) DeleteComment(context.Context, *documents.DeleteCommentRequest) (*emptypb.Empty, error) {
```

Comment:
Same as conversation...

Reply:
Same response :) Not implemented yet. And as a user I see no reason why can't I remove a comment. I'm able to delete documents, so I'd expect to do the same with comments. But there're many rabbit holes on how to handle that, so it's left unimplemented.